### ISSUE TITLE
the parser cannot read in more than approx. 32000 data points

### ISSUE THREAD

**Issue by moosler** [moosler/Sor-Viewer#1]
As soon as more than 32000 points are included, an error message is thrown
The error is currently being avoided by simply no longer reading out more data points

**Comment by vv3** [comment]
I took a stab at debugging this. It seems there is an overflow problem
in the code. I don't think it is related to a specific limit of how
many data points there is in the file. Whether this is a parsing error
in the code, or there is an error in the SOR file itself, I dont
know. Either way, the error can be masked by a workaround.

Example: I observe in the readVal function that this.data has 75296
bytes, but position=75296 and end=75298. This means that the slice
function "overflows" the size of this.data and returns an empty array
which later causes the exception to be thrown.

I have implemented an alternative workaround at: 
https://github.com/vv3/Sor-Viewer/commit/ab8af550988a4e1293fab665e474a4ef91e44b41
I will test this further and maybe send a Pull Request.
